### PR TITLE
Auth: Add support for role mapping and allowed groups in Google OIDC

### LINKED ISSUE #72797: Auth: Add ability to sync org roles from Google OAuth provider setup
**context**
We do not have the ability to sync org roles from the Google  OAuth provider currently. We have options affecting the org role that does not get implemented currently due to the fact that we do not support syncing role from their groups.

**proposals:**
1. nothing - if users have issues, tell them to use skip_org_role_sync to enable manual role updates;
2. add an exception to role locking for Google;
3. implement role mapping for Google.

Decision was made to set the `skip_org_role_sync = true` for Google until we have the ability to sync groups (to org roles) from Google aiming for 10.1. For 10.2

**DOD**

```[tasklist]
### Tasks
- [ ] https://github.com/grafana/grafana/issues/72816
- [ ] https://github.com/grafana/grafana/issues/72824
- [ ] https://github.com/grafana/grafana/issues/72869
- [ ] https://github.com/grafana/grafana/issues/72870
- [ ] https://github.com/grafana/grafana/issues/72871
```

### LINKED ISSUE #72824: Auth: OAuth Google implement org role sync

### LINKED ISSUE #72869: Auth: OAuth Google Enforce Sync

### LINKED ISSUE #72870: Auth: OAuth Google Role Mapping

### LINKED ISSUE #72871: Auth: OAuth Google GrafanaAdmin mapping

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

Add support for the following oauth config fields
```
[auth.google]
allowed_groups = ...
role_attribute_strict = ...
role_attribute_path = ...
```

Closes https://github.com/grafana/grafana/issues/72797
Closes https://github.com/grafana/grafana/issues/72824
Closes https://github.com/grafana/grafana/issues/72869
Closes https://github.com/grafana/grafana/issues/72870
Closes https://github.com/grafana/grafana/issues/72871

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by kalleep — APPROVED** [review]
Looks good, left one comment but it is not a blocker

File: pkg/login/social/gitlab_oauth.go [hunk 1]
```diff
@@ -22,7 +22,6 @@ const (
 
 type SocialGitlab struct {
 	*SocialBase
-	allowedGroups   []string
```
[the remark this answers is not in the document]
Comment (Jguer) [reply pkg/login/social/gitlab_oauth.go:25]: forgot to remove that one :sweat_smile:

**Review by IevaVasiljeva — APPROVED** [review]
Looks good (although I didn't test it myself). Thanks for getting it in so fast!

File: pkg/setting/setting.go [hunk 2]
```diff
@@ -1497,9 +1497,7 @@ func readAuthGithubSettings(cfg *Cfg) {
 func readAuthGoogleSettings(cfg *Cfg) {
 	sec := cfg.SectionWithEnvOverrides("auth.google")
 	cfg.GoogleAuthEnabled = sec.Key("enabled").MustBool(false)
-	// FIXME: for now we skip org role sync for google auth
-	// as we do not sync organization roles from Google
-	cfg.GoogleSkipOrgRoleSync = true
+	cfg.GoogleSkipOrgRoleSync = sec.Key("skip_org_role_sync").MustBool(false)
```
Comment (gamab) [inline pkg/setting/setting.go:1500]: This contradicts with the `defaults.ini`:
`skip_org_role_sync = true`

File: docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md [hunk 3]
```diff
@@ -132,3 +122,66 @@ With team sync, you can easily add users to teams by utilizing their Google grou
...
+```
+
+> Note: Add the `https://www.googleapis.com/auth/cloud-identity.groups.readonly` scope to your Grafana `[auth.google]` scopes configuration to retrieve groups
+
+#### Map server administrator role
+
+In this example, the user with email `admin@company.com` has been granted the `Admin` organization role as well as the Grafana server admin role.
+All other users are granted the `Viewer` role.
+
+```bash
```
Comment (gamab) [inline docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md:185]:
```suggestion
```ini
```

Comment (gamab) [reply docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md:185]: I think we should also advice them to set the `allow_assign_grafana_admin` given it defaults to false:
```suggestion
```ini
allow_assign_grafana_admin = true
```

**Review by gamab — APPROVED** [review]
Very good job :tada:
I left two suggestions for the docs :+1:

File: pkg/setting/setting.go (hunk 2 above)
Comment (Jguer) [inline pkg/setting/setting.go:1500]:
```suggestion
	cfg.GoogleSkipOrgRoleSync = sec.Key("skip_org_role_sync").MustBool(true)
```

**Review by chri2547 — APPROVED** [review]

**Pull request merged**